### DISCUSSION TITLE
Does GH reject commits if there is no change in code?

### DISCUSSION THREAD

**Discussion by Nithur-M** [octokit/octokit.js#2254] [Q&A] [answered]
I am using octokitjs to commit to a repo. But sometimes files aren't committed. The code works fine. Does GitHub reject commits if there is no change?

**Comment by timrogers** [comment]
@Nithur-M Hi there! Could you please share?:

* the code you’re using to create the commit (just the line that used Octokit should be fine!
* the response from the API that you get when the file isn’t committed (you can just use `console.log`)

That’ll help us to figure out what’s going wrong.

**Comment by Nithur-M** [comment, marked as answer]
Hi, thanks for responding. I think GH automatically doesn't write the commits if there aren't any changes in the files. I intentionally changed the content and committed it again, it works fine.
